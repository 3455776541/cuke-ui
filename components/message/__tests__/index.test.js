import React from "react";
import assert from "power-assert";
import { render, shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Message from "../index";

describe("<Message/>", () => {
  it("should render a <Message/> components", () => {
    const wrapper = render(
      <div>
        <Message title="哈哈" type="success" />
        <Message title="哈哈" type="error" />
        <Message title="哈哈" type="info" />
        <Message title="哈哈" type="warning" />
        <Message title="哈哈" type="loading" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-turn-table classnames", () => {
    const wrapper = shallow(<Message title="哈哈" type="success" />);
    assert(wrapper.find(".cuke-message").length === 1);
    assert(wrapper.find(".message-success").length === 1);
  });

  it("should 2s ago emit callback", () => {
    const wrapper = mount(
      <Message
        title="哈哈"
        type="success"
        duration={2}
        onClose={() => wrapper.setProps({ title: "回调" })}
      />
    );

    setTimeout(() => {
      assert(wrapper.props().title === "回调");
    }, 2000);
  });
});
