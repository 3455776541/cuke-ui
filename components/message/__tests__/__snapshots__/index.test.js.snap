// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message/> should render a <Message/> components 1`] = `
<div>
  <div
    class="cuke-message open"
  >
    <div
      class="cuke-message-title-custom message-success"
    >
      <p
        class="icon"
      >
        <svg
          attr="[object Object]"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
          />
          <polyline
            points="22 4 12 14.01 9 11.01"
          />
        </svg>
      </p>
      <p
        class="text"
      >
        <span
          class="title"
        >
          哈哈
        </span>
      </p>
    </div>
  </div>
  <div
    class="cuke-message open"
  >
    <div
      class="cuke-message-title-custom message-error"
    >
      <p
        class="icon"
      >
        <svg
          attr="[object Object]"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="15"
            x2="9"
            y1="9"
            y2="15"
          />
          <line
            x1="9"
            x2="15"
            y1="9"
            y2="15"
          />
        </svg>
      </p>
      <p
        class="text"
      >
        <span
          class="title"
        >
          哈哈
        </span>
      </p>
    </div>
  </div>
  <div
    class="cuke-message open"
  >
    <div
      class="cuke-message-title-custom message-info"
    >
      <p
        class="icon"
      >
        <svg
          attr="[object Object]"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <polygon
            points="11 5 6 9 2 9 2 15 6 15 11 19 11 5"
          />
          <path
            d="M19.07 4.93a10 10 0 0 1 0 14.14M15.54 8.46a5 5 0 0 1 0 7.07"
          />
        </svg>
      </p>
      <p
        class="text"
      >
        <span
          class="title"
        >
          哈哈
        </span>
      </p>
    </div>
  </div>
  <div
    class="cuke-message open"
  >
    <div
      class="cuke-message-title-custom message-warning"
    >
      <p
        class="icon"
      >
        <svg
          attr="[object Object]"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="12"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="16"
          />
        </svg>
      </p>
      <p
        class="text"
      >
        <span
          class="title"
        >
          哈哈
        </span>
      </p>
    </div>
  </div>
  <div
    class="cuke-message open"
  >
    <div
      class="cuke-message-title-custom message-loading"
    >
      <p
        class="icon"
      >
        <svg
          attr="[object Object]"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="12"
            x2="12"
            y1="2"
            y2="6"
          />
          <line
            x1="12"
            x2="12"
            y1="18"
            y2="22"
          />
          <line
            x1="4.93"
            x2="7.76"
            y1="4.93"
            y2="7.76"
          />
          <line
            x1="16.24"
            x2="19.07"
            y1="16.24"
            y2="19.07"
          />
          <line
            x1="2"
            x2="6"
            y1="12"
            y2="12"
          />
          <line
            x1="18"
            x2="22"
            y1="12"
            y2="12"
          />
          <line
            x1="4.93"
            x2="7.76"
            y1="19.07"
            y2="16.24"
          />
          <line
            x1="16.24"
            x2="19.07"
            y1="7.76"
            y2="4.93"
          />
        </svg>
      </p>
      <p
        class="text"
      >
        <span
          class="title"
        >
          哈哈
        </span>
      </p>
    </div>
  </div>
</div>
`;
